code/game/objects/items/weapons/RCD.dm

```diff
@@ -144,239 +218,229 @@ RCD
+				AT.ChangeTurf(/turf/simulated/floor/plating)
+				return TRUE
+		return FALSE
+
+	if(isfloorturf(A))
+		if(checkResource(5, user))
+			to_chat(user, "Deconstructing Floor...")
+			playsound(loc, 'sound/machines/click.ogg', 50, 1)
+			if(do_after(user, 50 * toolspeed, target = A))
+				if(!useResource(5, user))
+					return FALSE
+				playsound(loc, usesound, 50, 1)
+				var/turf/AT = A
+				AT.ChangeTurf(/turf/space)
+				return TRUE
+		return FALSE
+
+	if(istype(A, /obj/machinery/door/airlock))
+		if(checkResource(20, user))
+			to_chat(user, "Deconstructing Airlock...")
+			playsound(loc, 'sound/machines/click.ogg', 50, 1)
+			if(do_after(user, 50 * toolspeed, target = A))
+				if(!useResource(20, user))
+					return FALSE
+				playsound(loc, usesound, 50, 1)
+				qdel(A)
+				return TRUE
+		return FALSE
+
+	if(istype(A, /obj/structure/window)) // You mean the grille of course, do you?
+		A = locate(/obj/structure/grille) in A.loc
+	if(istype(A, /obj/structure/grille))
+		if(!checkResource(2, user))
+			return 0
+		to_chat(user, "Deconstructing window...")
+		playsound(loc, 'sound/machines/click.ogg', 50, 1)
+		if(!do_after(user, 20 * toolspeed, target = A))
+			return 0
+		if(!useResource(2, user))
+			return 0
+		playsound(loc, usesound, 50, 1)
+		var/turf/T1 = get_turf(A)
+		QDEL_NULL(A)
+		for(var/obj/structure/window/W in T1.contents)
+			qdel(W)
+		for(var/cdir in cardinal)
+			var/turf/T2 = get_step(T1, cdir)
+			if(locate(/obj/structure/window/full/shuttle) in T2)
+				continue // Shuttle windows? Nah. We don't need extra windows there.
+			if(!(locate(/obj/structure/grille) in T2))
+				continue
+			for(var/obj/structure/window/W in T2)
+				if(W.dir == turn(cdir, 180))
+					qdel(W)
+			var/obj/structure/window/reinforced/W = new(T2)
+			W.dir = turn(cdir, 180)
+		return TRUE
+	return FALSE
+
+/obj/item/rcd/proc/mode_window(atom/A, mob/user)
+	if(isfloorturf(A))
+		if(locate(/obj/structure/grille) in A)
+			return 0 // We already have window
+		if(!checkResource(2, user))
+			return 0
+		to_chat(user, "Constructing window...")
+		playsound(loc, 'sound/machines/click.ogg', 50, 1)
+		if(!do_after(user, 20 * toolspeed, target = A))
+			return 0
+		if(locate(/obj/structure/grille) in A)
+			return 0 // We already have window
+		if(!useResource(2, user))
+			return 0
+		playsound(loc, usesound, 50, 1)
+		new /obj/structure/grille(A)
+		for(var/obj/structure/window/W in A)
+			qdel(W)
+		for(var/cdir in cardinal)
+			var/turf/T = get_step(A, cdir)
+			if(locate(/obj/structure/grille) in T)
+				for(var/obj/structure/window/W in T)
+					if(W.dir == turn(cdir, 180))
+						qdel(W)
+			else // Build a window!
+				var/obj/structure/window/reinforced/W = new(A)
+				W.dir = cdir
+		var/turf/AT = A
+		AT.ChangeTurf(/turf/simulated/floor/plating) // Platings go under windows.
+		return 1
 
 /obj/item/rcd/afterattack(atom/A, mob/user, proximity)
-	if(!proximity) return
-	if(istype(A,/area/shuttle)||istype(A,/turf/space/transit))

```

Comment:
It's worth noting that I did not readd `istype(A, /area/shuttle)` because it has never fucking worked. You can't click on a damn area, but it's checking the type of the atom you clicked.